File: WebFt30/src/app/core/store/models/ft30/port-with-trace-dto.ts
import { FiberState } from './ft-enums';
import { PortOfOtau } from './port-of-otau';
import { ReturnCode } from './return-code';

export class PortWithTraceDto {
  traceId!: string;
  portOfOtau!: PortOfOtau;
  lastTraceState!: FiberState;
  lastRtuAccidentOnTrace!: ReturnCode;
}

File: Backend/src/Iit/Iit.Fibertest.Dto/Underlying/PortWithTraceDto.cs
using System.Runtime.Serialization;

namespace Iit.Fibertest.Dto
{
    [Serializable]
    [DataContract]
    public class PortWithTraceDto
    {
        [DataMember] public OtauPortDto OtauPort { get; set; } = null!;

        [DataMember]
        public Guid TraceId { get; set; }

        [DataMember]
        public FiberState LastTraceState { get; set; }


        // ReturnCode.MeasurementEndedNormally  if all is OK
        [DataMember]
        public ReturnCode LastRtuAccidentOnTrace { get; set; }
    }
}